openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/FairGBMDescriptorUtil.java

```diff
@@ -0,0 +1,200 @@
+                            + "(false positives).",
+                    NOT_MANDATORY,
+                    new ChoiceFieldType(ImmutableSet.of("FPR", "FNR", "FPR,FNR"), "FPR")
+            ),
+
+            // Parameters related to global constraints
+            new ModelParameter(
+                    "global_constraint_type",
+                    "(Fairness) Global constraint type",
+                    "Type of global constraint to enforce during training of fairness constraints. "
+                            + "For instance, if you want to deploy your model on 5% FPR you should use a global "
+                            + "constraint on 5% FPR, in order for the fairness constraints to be trained with this "
+                            + "FPR constraint in mind. Otherwise, fairness may not generalize well when you change "
+                            + "the model's operating point.",
+                    NOT_MANDATORY,
+                    new ChoiceFieldType(ImmutableSet.of("FPR", "FNR", "FPR,FNR"), "FPR,FNR")
+            ),
+            new ModelParameter(
+                    "global_target_fpr",
+                    "(Fairness) Global target FPR",
+                    "This is an inequality constraint: inactive when FPR is lower than the target. "
+                            + "If you have some (approximate) value of FPR in mind for deploying the model, then "
+                            + "set it here as well so that fairness constraints can better adapt to that value. "
+                            + "Oftentimes, some tension is required between global FPR and FNR constraints in order to "
+                            + "achieve the target values (a global constraint on FPR and FNR simultaneously).",
+                    NOT_MANDATORY,
+                    doubleRange(0.0, 1.0, 0.05)
+            ),
+            new ModelParameter(
+                    "global_target_fnr",
+                    "(Fairness) Global target FNR",
+                    "This is an inequality constraint: inactive when FNR is lower than the target. "
+                            + "If you have some (approximate) value of FNR in mind for deploying the model, then "
+                            + "set it here as well so that fairness constraints can better adapt to that value. "
+                            + "Oftentimes, some tension is required between global FPR and FNR constraints in order to "
+                            + "achieve the target values (a global constraint on FPR and FNR simultaneously).",
+                    NOT_MANDATORY,
+                    doubleRange(0.0, 1.0, 0.5)
+            ),
+
+            new ModelParameter(
+                    "objective",
+                    "(Fairness) Objective function",
+                    "For FairGBM you must use a constrained optimization function. "
+                            + "Currently, `constrained_cross_entropy` is recommended for most cases.",
```

Comment:
```suggestion
                            + "`constrained_cross_entropy` is recommended for most cases.",
```

Reply:
Do we know what is recommended in other cases? Should we add so that users understand the options?

Reply:
hmm, right now only `constrained_cross_entropy` is to be used in practice, but other "experimental" loss functions are supported (more for research purposes)